pkg/apis/api.acorn.io/v1/types.go

```diff
@@ -341,7 +341,8 @@ type Info struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-	Spec InfoSpec `json:"spec,omitempty"`
+	Regions   map[string]InfoSpec `json:"regions,omitempty"`
```

Comment:
Could you elaborate on this? Just want to understand.

My opinion is that adding this named typed makes the code less readable because I don't know what the type InfoSpecs is. I would have to look at it to see that it is just a map in disguise.

That is just my opinion though. It could be wrong and am interested to hear more about yours.

Reply:
> Could you elaborate on this? Just want to understand.

This is mostly subjective, but in my opinion the extra types:
- serve as a way to better codify the semantics of the map key
- have slightly more utility since they can receive helper methods
- are a bit less clunky to use; i.e. the maps type literal doesn't proliferate
  ```go
    func foo(specs InfoSpecs) { //...
    // vs.
    func foo(specs map[string]InfoSpec) { //...

    specs := make(InfoSpecs, 10)
    // vs.
    specs := make(map[string]InfoSpec, 10)
  ```

> My opinion is that adding this named typed makes the code less readable because I don't know what the type InfoSpecs is. I would have to look at it to see that it is just a map in disguise.

Yeah, that's a great point. Now that you mentioned it, I can see how the extra layers of indirection could be misleading and I'd rather not introduce a footgun for my future self.

Reply:
Thank you for including your thoughts here. I think they are definitely valid reasons for including a type.